msu/hooks/skills/skill_container.nut

```diff
@@ -27,6 +27,74 @@
 		this.m.ScheduledChangesSkills.clear();
 	}
 
+	local add = o.add;
+	o.add = function( _skill, _order = 0 )
+	{
+		if (!_skill.isStacking())
+		{
+			local incrementAddedStack = function( _alreadyPresentSkill, _skillToAdd )
+			{
+				_alreadyPresentSkill.m.MSU_AddedStack += 1;
+				if (!::MSU.isNull(_skillToAdd.getItem()))
+				{
+					_alreadyPresentSkill.setItem(_skillToAdd.getItem());
+					foreach (i, itemSkill in _skillToAdd.getItem().m.SkillPtrs)
+					{
+						if (itemSkill.getID() == _skillToAdd.getID())
+						{
+							_skillToAdd.getItem().m.SkillPtrs[i] = _alreadyPresentSkill;
+							_skillToAdd.setItem(null);
+							break;
+						}
+					}
+				}
+
+				if (_alreadyPresentSkill.m.MSU_AddedStack > 0)
+				{
+					_alreadyPresentSkill.m.IsGarbage = false;
+				}
```

Comment:
When does this come up?

Reply:
I've encountered this during `skill_container` deserialization whereby if a skill is added whose `onAdded` function unequips and equips the mainhand item. If **only one** such skill is present then there is no problem, but if multiple such skills are present then the skills of the weapon will be set to `Garbage` when the weapon is removed a second time. This happens even while the skills reside within `SkillsToAdd` array. See my bug report on vanilla forum from PR #198.